src/components/VCSMenu.tsx

```diff
@@ -80,22 +87,115 @@ export class VCSMenu extends React.Component<VCSMenuProps, VCSMenuState> {
     this.updateSelectedVariables = this.updateSelectedVariables.bind(this);
   }
 
+  async resetState() {
+    console.log("VCS Menu state reset.");
+    this.graphicsMenuRef.resetGraphicsState();
+    this.varMenuRef.setSelected(new Array<Variable>());
+    this.setState({
+      file_path: "",
+      plotReady: false,
+      loadedVariables: new Array<Variable>(),
+      selectedVariables: new Array<Variable>(),
+      selected_gm_group: "",
+      selected_gm: "",
+      selected_template: ""
+    });
+  }
+
   update(vars: Array<string>, gms: Array<any>, templates: Array<any>) {
     console.log(vars, gms, templates);
   }
 
+  getGraphicsSelections() {
+    // Load the selected graphics method from meta data (if exists)
+    let gm_data: [string, string] = notebook_utils.getMetaDataNow(
+      this.state.notebook_panel,
+      GRAPHICS_METHOD_KEY
+    );
+
+    // If the data is not null, set the selected graphic method and group
+    if (gm_data) {
```

Comment:
For code patterns like this where you call a method, check if its return is null and do something I think its a bit more clear to do:

```
value = someFunction(arg1, arg2)

if value is Null:
    reset()
    return

handle_value1(value)
handle_value2(value)
```

Its the same thing, but doing to null case first lets you exit early and avoid the else clause (because you return in the is null case), which cuts down on excessive brackets inside brackets inside brackets inside brackets which JS really suffers from.